### ISSUE TITLE
When request is blocked by a user filter rule, unblock should uncheck/delete it

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#1380]
We now allow the user to add a new "unblocking" rule.

Instead, for user filter rule, we could offer more options:

1. Unblock with a rule
2. Disable user filter rule
3. Delete user filter rule

Not sure about disabling/deleting, I guess we should stick to a single thing (either deleting or disabling)

TODO: We should re-think this, see the comment: https://github.com/AdguardTeam/AdguardForAndroid/issues/1380#issuecomment-440283180

**Comment by TPS** [comment]
This is particularly awful in latest v3ß.  When I press "Unblock" I want create a `@@` rule, not delete my blocking rule!  A _menu_ presenting _additional_ options is ok!

**Comment by TPS** [comment]
I'm the user, right?  If I want to spam my own user filterlist, let me.  But y'all might _suggest_ it'd be better to delete/disable/modify the rule in a particular fashion than forcing it.

Also, if complexity of that user list is a problem, maybe _suggest_ ways to trim if it hits a threshold (say, 100 rules — I'm 4× that already 😅), like deleting rules that duplicate others, especially from subscribed filterlists.

**Comment by ameshkov** [comment]
The problem with the old approach is simple.

1. Imagine you block an URL;
2. Then, sometime later you unblock it via adding a new unblocking rule to the user filter;
3. And next time when you want to block it back, you simply cannot because you already have an unblocking rule in the user filter;

The only solution is to modify user filter rules, not create new ones.

On the other hand, there's a new issue introduced by this. You can block the whole `example.org` and would like to unblock `example.org/good-stuff/*` only.

@admitrevskiy let's rollback to the previous behavior for now.

**Comment by TPS** [comment]
Thanks!  I agree, that is a problem.  Maybe keep old behavior as advanced menu toggle, & variation of new behavior as newbie default?

**Comment by artemiv4nov** [comment]
Well, this behavior has been improved in the 4.0 version, please follow updates.
